whispering/utils/path_ops.py

```diff
@@ -44,13 +44,8 @@ def get_files_with_extension(extension, directory='.'):
     :return: A list of all the files with the given extension.
     """
     # First, get all the files in the directory:
-    files = glob.glob(directory + '/*')
-    # Now go through all the files and add the ones with the given extension to the list:
-    files_with_extension = []
-    for file in files:
-        if file.endswith(extension):
-            files_with_extension.append(file)
-    return files_with_extension
+    files = glob.glob(f'{directory}/*')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;directory&#39; (index: 1 | type: Unknown). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;extension&#39;. The same parameter type is validated here for example: ai_sent/code_search_two.py:19. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).